store/src/server/api.rs

```diff
@@ -17,23 +21,33 @@ impl store::api_server::Api for DBApi {
     async fn check_nullifiers(
         &self,
         _request: tonic::Request<Streaming<store::CheckNullifiersRequest>>,
-    ) -> Result<Response<Self::CheckNullifiersStream>, Status>
-    {
+    ) -> Result<Response<Self::CheckNullifiersStream>, Status> {
         todo!()
     }
 }
 
-pub async fn serve(config: StoreConfig) -> Result<()> {
+pub async fn serve(config: StoreConfig, mut pool: Pool) -> Result<()> {
     let host_port = (config.endpoint.host.as_ref(), config.endpoint.port);
     let addrs: Vec<_> = host_port.to_socket_addrs()?.collect();
 
+    let nullifiers = pool.get_nullifiers().await?;
+    let leaves = nullifiers.into_iter().map(|(nullifier, block)| {
+        (
+            nullifier,
+            [Felt::new(block), Felt::ZERO, Felt::ZERO, Felt::ZERO],
+        )
+    });
+    let nullifier_tree = TieredSmt::with_leaves(leaves)?;
+    let db = store::api_server::ApiServer::new(DBApi {
+        pool,
+        nullifier_tree,
+    });
```

Comment:
Would be great to collect/print out info on how long these steps take. For example, we could print out something like;
```
Nullifier tree loaded in 123 ms.
```
Or maybe even:
```
Nullifier tree loaded with 1000 values in 123 ms.
```

Reply:
what time do you want to measure? The load is execute in an async function that is scheduled in another block, I can print the wall-time, but that is not necessarily the time it takes to read the data.

also a note: the `tracing-subscriber` crate does not print the span timings by default. It is possible to add a layer to track it and print it as log lines, but the idea is usually to collect open telemetry data and publish that to a server like `Jaeger`.

For the time being I added a span and a count of the number of entries, let me know what you want for the timing metric.

Reply:
I think the most interesting part would be the time it took to construct the TSMT as I suspect that this will be the bottleneck. I think the code there is synchronous - so, wall time should be fine.

In general, I'm expecting that once the database gets to some real size, startup time will be non-negligible and we'll spend quite a bit of time on optimizing it - so, knowing which parts contribute to the start-up time would be beneficial.

Reply:
added time measurement for `TieredSmt::with_leaves`